### PR TITLE
Add authentication via SSO

### PR DESCRIPTION
**Pull request by Letiste**
This PR adds:
 - connection with the SSO system
 - login/logout logic
 - session expiration

### CODE REVIEW

File: docs/features/authentication.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Authentication
+
+The authentication is done via a SSO system.
+
+The general flow for the authentication is described below:
```
Comment (Letiste) [inline docs/features/authentication.md:5]: Preview for the diagram: https://github.com/rezoleo/lea5/blob/auth-sso/docs/features/authentication.md

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def log_in(user)
+    reset_session # For security reasons, we clear the session data before login
```
Comment (Letiste) [inline app/helpers/sessions_helper.rb:13]: As explained here: https://guides.rubyonrails.org/security.html#session-fixation-countermeasures

File: config/initializers/_constants.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
```
Comment (Letiste) [inline config/initializers/_constants.rb:1]: The file's name starts with an underscore to ensure the constants are loaded first

File: config/initializers/omniauth.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Rails.application.config.middleware.use OmniAuth::Builder do
+  if Rails.env.production?
+    client_options = {
+      identifier: ENV.fetch('SSO_PROD_ID'),
+      secret: ENV.fetch('SSO_PROD_SECRET'),
+      redirect_uri: "https://lea5.rezoleo.fr/#{AUTH_CALLBACK_PATH}"
+    }
+  elsif Rails.env.development?
+    client_options = {
+      identifier: ENV.fetch('SSO_DEV_ID'),
```
Comment (Letiste) [inline config/initializers/omniauth.rb:12]: With this way of fetching the env var, we need to set them up also when using the CLI (except for the tests). Meaning that they are required to generate controllers for instance, even if we don't need them.
If you have an alternative, I would gladly accept it

File: config/initializers/_constants.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+# Path to login via SSO authentication
+AUTH_PATH = '/auth/keycloak'
+
+# Callback path from SSO authentication
+AUTH_CALLBACK_PATH = "#{AUTH_PATH}/callback".freeze
+
+SESSION_DURATION_TIME = 3.hours
```
Comment (Letiste) [inline config/initializers/_constants.rb:9]: Not sure if it is the best way (or at least the rails way) to centralize this kind of information

File: config/initializers/_constants.rb (hunk 3 above)
Comment (Letiste) [reply config/initializers/_constants.rb:1]: Not sure if it is the best way (or at least the rails way) to centralize this kind of information

**Review by nymous — APPROVED** [review]
Reviewed together, and it's :fire:!

**Pull request merged**